Abstract Syntax and Semantics Document/index.html

```diff
@@ -500,53 +500,159 @@
+        for every stream <code>S</code>, 
+        an element <code>S(i)</code> of <code>S</code> is a member of <code>w(S)</code> if and only if
+        the timestamp predicate  <code>P</code> of <code>S(i)</code> is in <code>P</code> and
+        the timestamp <code>t</code> of <code>S(i)</code> is contained in <code>w<sub>J</sub>(P)</code>.
+      </p>
     
-    <p><code>x = (l,d)</code>, where <code>l ∈ N ∪ {∞}</code> and <code>d ∈ N</code>. The function <code>w<sub>τ</sub>(S,p,t,x)</code> returns the substream of S that contains all timestamped graphs associated with predicate <code>p</code> of the last <code>l</code> time units relative to a pivot time point <code>t'</code> derived from <code>t</code> and the step size <code>d</code> (Todo: MD: a figure to illustrate). We use <code>l = ∞</code> to take all previous timestamped graphs.</p>
-    
-    <p>Formally:</p>
-    
-    <p><code>w<sub>τ</sub>(S,p,t,(l,d)) = {((n,g),p,t<sub>1</sub>)∈ S ∣ t'-l &lt; t<sub>1</sub> ≤ t'}</code>, </p>
-    
-    <p>where <code>t'= ⌊t/d⌋ \cdot d</code></p>
-    
-    <h3>Count-based window functions</h3>
-    
-    <p><code>x = (l)</code>, where <code>l ∈ N</code>. The function <code>w<sub>#</sub>(S,p,t,x)</code> selects a substream <code>S<sub>1</sub></code> of <code>S</code> based on the time instant <code>t'</code> satisfying that:</p>
-    
-    <ul>
-      <li>for every <code>t' &lt; t'' ≤ t</code>, there are fewer than <code>l</code> timestamped graphs associated with predicate <code>p</code> in <code>S</code> from <code>t''</code> to <code>t</code>,</li>
-      <li>there are at least <code>l</code> timestamped graphs associated with predicate <code>p</code> in <code>S</code> from <code>t'</code> to <code>t</code>.</li>
-    </ul>
-    
-    <p>Elements from <code>S<sub>1</sub></code> are those <code>((n<sub>i</sub>,g<sub>i</sub>),p,t<sub>i</sub>)</code> from <code>S</code> having <code>t<sub>i</sub> ≥ t'</code>. In case there are more than <code>l</code> timestamped graphs associated with predicate <code>p</code> in <code>S</code> from <code>t'</code> to <code>t</code>, only timestamped graphs at <code>t'</code> are removed at random.</p>
-    
-    <p>Formally:</p>
-    
-    <p><code>w<sub>#</sub>(S,p,t,(l)) = {((n,g),p,t'') ∈ S ∣ t' &lt; t'' ≤ t} ∪ X(S|<sub>p</sub>[t',t'], l - #S|<sub>p</sub>[t'+1,t])</code>,</p>
-    
-    <p>where</p>
-    
-    <ul>
-      <li><code>S|<sub>p</sub>[t<sub>1</sub>,t<sub>2</sub>] = {((n,g),p,t'') ∈ S ∣ t<sub>1</sub> ≤ t'' ≤ t<sub>2</sub>}</code>,</li>
-      <li><code>#S|<sub>p</sub>[t<sub>1</sub>,t<sub>2</sub>]</code> is the number of elements of this set,</li>
-      <li><code>t'</code> satisfies that <code>#S|<sub>p</sub>[t',t] ≥ l</code> and <code>#S|<sub>p</sub>[t'+1,t] &lt; l</code> </li>
-    </ul>
+      <p class="note">The substream resulting from the application of a time-based window function is time-bounded.</p>
     </section>
```

Comment:
Yes, we will add (at least) two running examples, one leading into the time-series profile and one leading to the linked-list profile. I am waiting until we get these basic definitions stabilized and do some reorganization because otherwise the branches get tangled up.